engine/class_modules/sc_death_knight.cpp

```diff
@@ -9263,6 +9284,8 @@ double death_knight_t::composite_player_pet_damage_multiplier( const action_stat
     m *= 1.0 + cache.mastery_value();
   }
 
+  m *= 1.0 + get_target_data( state->target )->debuff.unholy_blight->stack_value();
```

Comment:
This will increase damage based on the state's target's (aka the spell's target's) stacks. 

You should move this to death_knight_pet_t::composite_player_target_multiplier().
Unless this also affect non DK pet damage in which case I'm not sure how to handle this.

Reply:
Why is the spell target not what we want?

Reply:
If you cast a single aoe ability, there will be a single state object that contains the target of the spell cast.

Reply:
You want to make sure that the modifier uses the right target's data, which is why you should move this to a target_multiplier, that will affect the damage done to a specific target. (composite_player_pet_damage_multiplier affects damage based on the source).